source/extensions/tracers/xray/tracer.cc

```diff
@@ -0,0 +1,124 @@
+#include "extensions/tracers/xray/tracer.h"
+
+#include <chrono>
+#include <ctime>
+#include <random>
+
+#include "common/common/utility.h"
+#include "common/tracing/http_tracer_impl.h"
+
+#include "extensions/tracers/xray/util.h"
+#include "extensions/tracers/xray/xray_core_constants.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+const std::string Tracer::VERSION_ = "1";
+const std::string Tracer::DELIMITER_ = "-";
```

Comment:
no static non-pods. See https://github.com/envoyproxy/envoy/blob/master/STYLE.md in general and https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use in particular.

In this case just declare as a const char[] and avoid the fiasco.

Also I don't the naming of const member vars is addressed specifically in the style guide but in all uses I've seen they are named like non-const member variables (version_, delimeter_) rather than const non-member variables.

Reply:
I addressed it by using const char* and thanks for your suggestion.